backend/kesaseteli/applications/jobs/daily/tests/test_clean_youth_applications.py

```diff
@@ -0,0 +1,162 @@
+    date_less_than_year_ago = timezone.now() - relativedelta(years=1, days=-1)
+
+    # Create different applications
+    rejected_applications_created_year_ago = set_creation_date(
+        RejectedYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    rejected_applications_created_less_than_year_ago = set_creation_date(
+        RejectedYouthApplicationFactory.create_batch(size=5),
+        created_at=date_less_than_year_ago,
+    )
+
+    unhandled_applications_created_year_ago = set_creation_date(
+        UnhandledYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    accepted_applications_created_year_ago = set_creation_date(
+        AcceptedYouthApplicationFactory.create_batch(size=5), created_at=date_year_ago
+    )
+
+    # Run the batch job
+    clean_rejected_youth_applications.Job().execute()
+
+    # Rejected application older than year should have been cleaned
+    for application in rejected_applications_created_year_ago:
+        assert not YouthApplication.objects.filter(pk=application.id).exists()
+
+    # Newer rejected applications should still exist
+    for application in rejected_applications_created_less_than_year_ago:
+        assert YouthApplication.objects.filter(pk=application.id).exists()
+
+    # Unhandled applications should still exist
+    for application in unhandled_applications_created_year_ago:
+        assert YouthApplication.objects.filter(pk=application.id).exists()
+
+    # Accepted applications should still exist
+    for application in accepted_applications_created_year_ago:
+        assert YouthApplication.objects.filter(pk=application.id).exists()
+
+
+@pytest.mark.django_db
+def test_clean_all_youth_applications_older_than_five_years():
+    date_five_years_ago = timezone.now() - relativedelta(years=5)
+    date_under_five_years_ago = timezone.now() - relativedelta(years=5, days=-1)
+
+    # Create applications with all possible statuses
+    applications_created_five_years_ago = set_creation_date(
+        [
+            YouthApplicationFactory(status=status)
+            for status in YouthApplicationStatus.values
+        ],
+        created_at=date_five_years_ago,
+    )
+
+    applications_created_less_than_five_years_ago = set_creation_date(
+        [
+            YouthApplicationFactory(status=status)
+            for status in YouthApplicationStatus.values
+        ],
+        created_at=date_under_five_years_ago,
+    )
+
+    # related summer vouchers
+    summer_vouchers_created_five_years_ago = YouthSummerVoucher.objects.filter(
+        youth_application__in=[app.pk for app in applications_created_five_years_ago]
+    )
+    assert len(summer_vouchers_created_five_years_ago) > 0
+
+    summer_vouchers_created_less_than_five_years_ago = (
+        YouthSummerVoucher.objects.filter(
+            youth_application__in=[
+                app.pk for app in applications_created_less_than_five_years_ago
+            ]
```

Comment:
No need to unpack the primary keys into a list, same thing as [previous comment](https://github.com/City-of-Helsinki/yjdh/pull/1582/commits/cc001f534a92c97dbb02b299ae09c7d5ddf53dee#r1044123472).

Reply:
fixed and also cleaned up the assetion for loops